src/identify/identify.ts

```diff
@@ -297,7 +300,7 @@ export class DefaultIdentifyService implements Startable {
     }
   }
 
-  async identify (connection: Connection, options: AbortOptions = {}): Promise<void> {
+  async identify (connection: Connection, options: AbortOptions = {}): Promise<IdentifyResult> {
```

Comment:
Please can you create an `IdentifyService` interface in the index file that contains only this method, have the factory function export it and have this class implement it.

This looks like something that was missed, oops.

The interface method could also use a JSDoc that says something like "due to the default limits on inbound/outbound streams for this protocol, invoking this method when `runOnConnectionOpen` is true can lead to unpredictable results as streams may be closed by the local or the remote node.  Please use with caution.  If you find yourself needing to call this method to discover other peers that support your protocol, you may be better off configuring a topology to be notified instead." and linking to the appropriate types.